core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -946,9 +937,8 @@ private object RTS {
     }
 
     def changeError[E1, E2, A](f: E2 => ExitResult[E1, A], cb: Callback[E1, A]): Callback[E2, A] = {
-      case ExitResult.Completed(a)  => cb(ExitResult.Completed(a))
-      case ExitResult.Terminated(t) => cb(ExitResult.Terminated(t))
-      case ExitResult.Failed(e2)    => cb(f(e2))
+      case ExitResult.Failed(e2) => cb(f(e2))
+      case x                     => cb(x.asInstanceOf[ExitResult[E1, A]])
```

Comment:
This looks good, but maybe we can put this onto `ExitResult` to hide the unsafety? A `mapError` function, perhaps?